applications/asset_tracker_v2/src/modules/data_module.c

```diff
@@ -1430,43 +1405,47 @@ static void on_all_states(struct data_msg_data *msg)
-			     sizeof(msg->module.location.data.neighbor_cells.cell_data));
-
-		BUILD_ASSERT(sizeof(neighbor_cells.neighbor_cells) ==
-			     sizeof(msg->module.location.data.neighbor_cells.neighbor_cells));
+	if (IS_EVENT(msg, location, LOCATION_MODULE_EVT_CLOUD_LOCATION_DATA_READY)) {
+		if (msg->module.location.data.cloud_location.neighbor_cells_valid) {
+			BUILD_ASSERT(sizeof(cloud_location.neighbor_cells.cell_data) ==
+				     sizeof(msg->module.location.data.cloud_location
+					.neighbor_cells.cell_data));
 
-		memcpy(&neighbor_cells.cell_data,
-		       &msg->module.location.data.neighbor_cells.cell_data,
-		       sizeof(neighbor_cells.cell_data));
+			BUILD_ASSERT(sizeof(cloud_location.neighbor_cells.neighbor_cells) ==
+				     sizeof(msg->module.location.data.cloud_location
+					.neighbor_cells.neighbor_cells));
 
-		memcpy(&neighbor_cells.neighbor_cells,
-		       &msg->module.location.data.neighbor_cells.neighbor_cells,
-		       sizeof(neighbor_cells.neighbor_cells));
-
-		neighbor_cells.ts = msg->module.location.data.neighbor_cells.timestamp;
-		neighbor_cells.queued = true;
-
-		requested_data_status_set(APP_DATA_LOCATION);
-	}
+			cloud_location.neighbor_cells_valid = true;
+			memcpy(&cloud_location.neighbor_cells.cell_data,
+			       &msg->module.location.data.cloud_location.neighbor_cells.cell_data,
+			       sizeof(cloud_location.neighbor_cells.cell_data));
 
+			memcpy(&cloud_location.neighbor_cells.neighbor_cells,
+			       &msg->module.location.data.cloud_location
+					.neighbor_cells.neighbor_cells,
+			       sizeof(cloud_location.neighbor_cells.neighbor_cells));
+		}
 #if defined(CONFIG_LOCATION_METHOD_WIFI)
-	if (IS_EVENT(msg, location, LOCATION_MODULE_EVT_WIFI_ACCESS_POINTS_DATA_READY)) {
-		BUILD_ASSERT(sizeof(wifi_access_points.ap_info) ==
-			     sizeof(msg->module.location.data.wifi_access_points.ap_info));
-
-		memcpy(&wifi_access_points.ap_info,
-		       &msg->module.location.data.wifi_access_points.ap_info,
-		       sizeof(wifi_access_points.ap_info));
-
-		wifi_access_points.cnt = msg->module.location.data.wifi_access_points.cnt;
-		wifi_access_points.ts = msg->module.location.data.wifi_access_points.timestamp;
-		wifi_access_points.queued = true;
+		if (msg->module.location.data.cloud_location.wifi_access_points_valid) {
+			BUILD_ASSERT(sizeof(cloud_location.wifi_access_points.ap_info) ==
+				     sizeof(msg->module.location.data.cloud_location
+					.wifi_access_points.ap_info));
+
+			cloud_location.wifi_access_points_valid = true;
+			memcpy(&cloud_location.wifi_access_points.ap_info,
+			       &msg->module.location.data.cloud_location
+					.wifi_access_points.ap_info,
+			       sizeof(cloud_location.wifi_access_points.ap_info));
+
+			cloud_location.wifi_access_points.cnt =
+				msg->module.location.data.cloud_location.wifi_access_points.cnt;
+		}
+#endif
+		cloud_location.ts = msg->module.location.data.cloud_location.timestamp;
+		cloud_location.queued = true;
```

Comment:
Need to set `ts` and `queued` for `cloud_location.neighbor_cells` and `cloud_location.wifi_access_points`

Reply:
Done. We could get rid of some queued flags in the future when WiFi support for AWS, Azure and Lwm2m is made.